src/ipmi.c

```diff
@@ -287,6 +325,13 @@ static int sensor_list_add(ipmi_sensor_t *sensor) {
   }
   } /* switch (sensor_type) */
 
+  /* if sensor provides the percentage value, add "_percent" suffix to the
+   * sensor collectd type */
+  if (ipmi_sensor_get_percentage(sensor)) {
+    ssnprintf(percent_type, sizeof(percent_type), "%s_percent", type);
```

Comment:
Please use the existing `percent` type instead of introducing more specialized `foo_percent` types. The original type should go into the type instance. For example, "power-ac0" (type: "power", type instance: "ac0") would become "percent-power-ac0" (type: "percent", type instance: "power-ac0").

It's best to think of collectd's types as a "unit label", if that makes sense.

Reply:
I dislike such a mixing of different *types* into one 'percent' type.
IMHO, personally I would not use such implementation in my systems.